make test_distributed gloo test use MultiProcessTestCase

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#36970 make test_distributed gloo test use MultiProcessTestCase**

We would like to move all distributed testing to use the existing
multiprocessing tooling defined in common_distributed.py. With this change, we
make `TestDistBackend` inherit from `MultiProcessTestCase` and enable fork mode
multiprocessing. In the next step, we can enable spawn mode for these tests
which will give us TSAN coverage.

Differential Revision: [D21146947](https://our.internmc.facebook.com/intern/diff/D21146947/)

rohan-varma committed:
[WIP] make test_distributed gloo test use MultiProcessTestCase

We would like to move all distributed testing to use the existing
multiprocessing tooling defined in common_distributed.py. With this change, we
make `TestDistBackend` inherit from `MultiProcessTestCase` and enable fork mode
multiprocessing. In the next step, we can enable spawn mode for these tests
which will give us TSAN coverage.

Differential Revision: [D21146947](https://our.internmc.facebook.com/intern/diff/D21146947/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[WIP] make test_distributed gloo test use MultiProcessTestCase"


We would like to move all distributed testing to use the existing
multiprocessing tooling defined in common_distributed.py. With this change, we
make `TestDistBackend` inherit from `MultiProcessTestCase` and enable fork mode
multiprocessing. In the next step, we can enable spawn mode for these tests
which will give us TSAN coverage.

Differential Revision: [D21146947](https://our.internmc.facebook.com/intern/diff/D21146947/)

[ghstack-poisoned]

pritamdamania87 reviewed (approved):
Looks good! Please fix the lint issues before landing.

rohan-varma committed:
Update on "make test_distributed gloo test use MultiProcessTestCase"


We would like to move all distributed testing to use the existing
multiprocessing tooling defined in common_distributed.py. With this change, we
make `TestDistBackend` inherit from `MultiProcessTestCase` and enable fork mode
multiprocessing. In the next step, we can enable spawn mode for these tests
which will give us TSAN coverage.

Differential Revision: [D21146947](https://our.internmc.facebook.com/intern/diff/D21146947/)

[ghstack-poisoned]

rohan-varma committed:
Update on "make test_distributed gloo test use MultiProcessTestCase"


We would like to move all distributed testing to use the existing
multiprocessing tooling defined in common_distributed.py. With this change, we
make `TestDistBackend` inherit from `MultiProcessTestCase` and enable fork mode
multiprocessing. In the next step, we can enable spawn mode for these tests
which will give us TSAN coverage.

Differential Revision: [D21146947](https://our.internmc.facebook.com/intern/diff/D21146947/)

[ghstack-poisoned]
